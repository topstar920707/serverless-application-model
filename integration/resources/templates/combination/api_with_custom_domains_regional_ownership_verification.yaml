Parameters:
  MyDomainName:
    Type: String
  MyDomainCertificate:
    Type: String
  MyDomainOwnershipVerificationCertificate:
    Type: String

Globals:
  Api:
    Domain:
      DomainName:
        Ref: MyDomainName
      CertificateArn:
        arn:aws:acm:us-east-1:830899278857:certificate/7c9b4aa7-4262-4b20-85c0-ec45d8d55fa6
      EndpointConfiguration: REGIONAL
      MutualTlsAuthentication:
        TruststoreUri: ${mtlsuri}
        TruststoreVersion: 0
      SecurityPolicy: TLS_1_2
      BasePath:
        - /get
        - /post
      Route53:
        HostedZoneId: Z1DTV8GVAVOHDR
      OwnershipVerificationCertificateArn:
        Fn::Sub: arn:aws:acm:${AWS::Region}:${AWS::AccountId}:certificate/798c74e5-dc55-48b6-b1d8-13d2d7b3d265

Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
        exports.handler = async (event) => {
          const response = {
            statusCode: 200,
            body: JSON.stringify('Hello from Lambda!'),
          };
          return response;
        };
      Handler: index.handler
      Runtime: nodejs12.x
      Events:
        ImplicitGet:
          Type: Api
          Properties:
            Method: Get
            Path: /get
        ImplicitPost:
          Type: Api
          Properties:
            Method: Post
            Path: /post
