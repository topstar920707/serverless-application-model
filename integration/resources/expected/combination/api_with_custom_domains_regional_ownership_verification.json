[
  { "LogicalResourceId":"MyFunctionImplicitGetPermissionProd", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"ServerlessRestApiDeployment", "ResourceType":"AWS::ApiGateway::Deployment" },
  { "LogicalResourceId":"ServerlessRestApipostBasePathMapping", "ResourceType":"AWS::ApiGateway::BasePathMapping" },
  { "LogicalResourceId":"ServerlessRestApi", "ResourceType":"AWS::ApiGateway::RestApi" },
  { "LogicalResourceId":"RecordSetGroupddfc299be2", "ResourceType":"AWS::Route53::RecordSetGroup" },
  { "LogicalResourceId":"ServerlessRestApiProdStage", "ResourceType":"AWS::ApiGateway::Stage" },
  { "LogicalResourceId":"ServerlessRestApigetBasePathMapping", "ResourceType":"AWS::ApiGateway::BasePathMapping" },
  { "LogicalResourceId":"ApiGatewayDomainNamec6418178e6", "ResourceType":"AWS::ApiGateway::DomainName" },
  { "LogicalResourceId":"MyFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyFunctionImplicitPostPermissionProd", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyFunctionRole", "ResourceType":"AWS::IAM::Role" }
]